src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -617,6 +665,58 @@ Optional<float[]> getFloatsFromContext(ParseContext context, int dimension) thro
         return Optional.of(array);
     }
 
+    // Returns an optional array of float values where each value in the vector is parsed as a float and validated
+    // if it is a finite number and within the fp16 range of [-65504 to 65504] by default.
+    // If the Index setting, "index.knn.faiss.clip_to_fp16_range" is set to True, if the vector value is
+    // outside the FP16 range then it will be clipped to FP16 range.
+    Optional<float[]> getFloatsFromContextWithFP16Validation(ParseContext context, int dimension) throws IOException {
+        context.path().add(simpleName());
+
+        ArrayList<Float> vector = new ArrayList<>();
+        XContentParser.Token token = context.parser().currentToken();
+        boolean clipToFP16Range = isFaissSQClipToFP16RangeEnabled(
+            (MethodComponentContext) fieldType().getKnnMethodContext()
+                .getMethodComponentContext()
+                .getParameters()
+                .get(METHOD_ENCODER_PARAMETER)
+        );
+        float value;
+        if (token == XContentParser.Token.START_ARRAY) {
```

Comment:
Could we refactor so code is shared between getFloatsFromContext and getFloatsFromContextWithFP16Validation?

Reply:
updated